File: manufacturing.py
# This Python file uses the following encoding: utf-8
import sys

from PySide6.QtWidgets import QApplication, QMainWindow
from Optimisers.ManufacturingQuestion import ManuPerf
# Important:
# You need to run the following command to generate the ui_form.py file
#     pyside6-uic form.ui -o ui_form.py, or
#     pyside2-uic form.ui -o ui_form.py

from ui_manufacturing import Ui_wManufacturing

class ManufacturingWindow(QMainWindow):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.ui = Ui_wManufacturing()
        self.ui.setupUi(self)

        self.ui.btnCalculate.clicked.connect(self.btnCalculate_clicked)
        self.ui.btnBack.clicked.connect(self.btnBack_clicked)
        print("Initialised")

    def btnCalculate_clicked(self):
        print("Clicked")
        ShowPlot = False
        if self.ui.cbPlot.isChecked():
            ShowPlot = True

        Function = str(self.ui.txtFunction.text())
        min = float(self.ui.txtMin.text())
        max = float(self.ui.txtMax.text())
        h = float(self.ui.txtStep.text())

        self.ui.txtOutput.setText(ManuPerf.manufacturing(Function, min, max, h, ShowPlot))


    def btnBack_clicked(self):
        from menu import MainWindow
        print("Back")
        self.hide()
        self.ui_menu = MainWindow()
        self.ui_menu.show()


if __name__ == "__main__":
    app = QApplication(sys.argv)
    widget = ManufacturingWindow()
    widget.show()
    sys.exit(app.exec())

File: manufacturing.cpp
// manufacturing.cpp
#include "manufacturing.h"
#include "menu.h"
#include "python_utils.h"       // our helper for calling into Python
#include <QDebug>

ManufacturingWindow::ManufacturingWindow(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::wManufacturing)
{
    ui->setupUi(this);  // build the widgets :contentReference[oaicite:2]{index=2}
    connect(ui->btnCalculate, &QPushButton::clicked,
            this, &ManufacturingWindow::btnCalculate_clicked);
    connect(ui->btnBack,      &QPushButton::clicked,
            this, &ManufacturingWindow::btnBack_clicked);
    qDebug() << "ManufacturingWindow initialized";
}

ManufacturingWindow::~ManufacturingWindow()
{
    delete ui;
}

void ManufacturingWindow::btnCalculate_clicked()
{
    qDebug() << "Calculate clicked";
    bool showPlot = ui->cbPlot->isChecked();
    QString fn    = ui->txtFunction->text();
    QString minV  = ui->txtMin->text();
    QString maxV  = ui->txtMax->text();
    QString step  = ui->txtStep->text();

    // Call the Python function ManuPerf.manufacturing(...)
    QStringList args = { fn, minV, maxV, step, showPlot ? "1" : "0" };
    QString result = callPythonFunction(
        "Optimisers.ManufacturingQuestion",  // module
        "manufacturing",                      // function name
        args
        );

    ui->txtOutput->setText(result);
}

void ManufacturingWindow::btnBack_clicked()
{
    qDebug() << "Back clicked";
    hide();
    Menu *m = new Menu(this);
    m->show();
}
